### ISSUE TITLE
irssi+http_proxy(squid) connection problems

### ISSUE THREAD

**Issue by nehaljwani** [irssi/irssi#145]
This is the scenario:
10.4.8.204 is the default gateway as well as the proxy server for the entire campus.
10.2.52.57 is the client trying to connect to irc.freenode.net:6667
An instance of squid runs on 10.4.8.204:8080 and accepts all http requests including CONNECT on port 6667.

Scenario1 (No proxy configured in irssi + forward rule on gateway):
When I add the following rule in the default gateway:
```
iptables -I FORWARD -i eth1 -p tcp -m tcp --dport 6667 -j ACCEPT
```
irssi works without any problems with the default config file formed after installation.
tcpdump:
```
[root@proxy ~]# tcpdump host 10.2.57.52 -nn
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
listening on eth1, link-type EN10MB (Ethernet), capture size 65535 bytes
01:32:55.640007 IP 10.2.57.52.65150 > 207.46.114.58.80: Flags [S], seq 804228678, win 8192, options [mss 1460,nop,wscale 8,nop,nop,sackOK], length 0
01:32:55.640061 IP 10.4.8.204 > 10.2.57.52: ICMP 207.46.114.58 tcp port 80 unreachable, length 60
01:32:56.880198 IP 10.2.57.52.46851 > 37.48.83.75.6667: Flags [S], seq 841079912, win 29200, options [mss 1460,sackOK,TS val 2689580076 ecr 0,nop,wscale 7], length 0
01:32:57.060468 IP 37.48.83.75.6667 > 10.2.57.52.46851: Flags [S.], seq 2861947565, ack 841079913, win 14480, options [mss 1460,sackOK,TS val 2137925891 ecr 2689580076,nop,wscale 6], length 0
01:32:57.060932 IP 10.2.57.52.46851 > 37.48.83.75.6667: Flags [.], ack 1, win 229, options [nop,nop,TS val 2689580257 ecr 2137925891], length 0
01:32:57.060980 IP 10.2.57.52.46851 > 37.48.83.75.6667: Flags [P.], seq 1:12, ack 1, win 229, options [nop,nop,TS val 2689580257 ecr 2137925891], length 11
01:32:57.241443 IP 37.48.83.75.6667 > 10.2.57.52.46851: Flags [.], ack 12, win 227, options [nop,nop,TS val 2137925937 ecr 2689580257], length 0
01:32:57.241454 IP 37.48.83.75.6667 > 10.2.57.52.46851: Flags [P.], seq 1:66, ack 12, win 227, options [nop,nop,TS val 2137925937 ecr 2689580257], length 65
01:32:57.241894 IP 10.2.57.52.46851 > 37.48.83.75.6667: Flags [P.], seq 12:59, ack 1, win 229, options [nop,nop,TS val 2689580438 ecr 2137925937], length 47
01:32:57.241925 IP 10.2.57.52.46851 > 37.48.83.75.6667: Flags [.], ack 66, win 229, options [nop,nop,TS val 2689580438 ecr 2137925937], length 0
01:32:57.422620 IP 37.48.83.75.6667 > 10.2.57.52.46851: Flags [P.], seq 66:186, ack 59, win 227, options [nop,nop,TS val 2137925982 ecr 2689580438], length 120
01:32:57.422946 IP 10.2.57.52.46851 > 37.48.83.75.6667: Flags [.], ack 186, win 229, options [nop,nop,TS val 2689580619 ecr 2137925982], length 0
01:32:58.639859 IP 10.2.57.52.65150 > 207.46.114.58.80: Flags [S], seq 804228678, win 8192, options [mss 1460,nop,wscale 8,nop,nop,sackOK], length 0
01:32:58.639882 IP 10.4.8.204 > 10.2.57.52: ICMP 207.46.114.58 tcp port 80 unreachable, length 60
01:33:04.655205 IP 10.2.57.52.65150 > 207.46.114.58.80: Flags [S], seq 804228678, win 8192, options [mss 1460,nop,nop,sackOK], length 0
01:33:04.655229 IP 10.4.8.204 > 10.2.57.52: ICMP 207.46.114.58 tcp port 80 unreachable, length 56
01:33:06.882473 IP 37.48.83.75.6667 > 10.2.57.52.46851: Flags [P.], seq 186:241, ack 59, win 227, options [nop,nop,TS val 2137928346 ecr 2689580619], length 55
01:33:06.882798 IP 10.2.57.52.46851 > 37.48.83.75.6667: Flags [.], ack 241, win 229, options [nop,nop,TS val 2689590079 ecr 2137928346], length 0
01:33:06.882978 IP 37.48.83.75.6667 > 10.2.57.52.46851: Flags [.], seq 241:1689, ack 59, win 227, options [nop,nop,TS val 2137928346 ecr 2689580619], length 1448
01:33:06.882986 IP 37.48.83.75.6667 > 10.2.57.52.46851: Flags [.], seq 1689:3137, ack 59, win 227, options [nop,nop,TS val 2137928346 ecr 2689580619], length 1448
01:33:06.883348 IP 37.48.83.75.6667 > 10.2.57.52.46851: Flags [.], seq 3137:4585, ack 59, win 227, options [nop,nop,TS val 2137928346 ecr 2689580619], length 1448
01:33:06.883519 IP 10.2.57.52.46851 > 37.48.83.75.6667: Flags [.], ack 1689, win 251, options [nop,nop,TS val 2689590080 ecr 2137928346], length 0
01:33:06.883601 IP 10.2.57.52.46851 > 37.48.83.75.6667: Flags [.], ack 3137, win 274, options [nop,nop,TS val 2689590080 ecr 2137928346], length 0
01:33:06.883725 IP 10.2.57.52.46851 > 37.48.83.75.6667: Flags [P.], seq 59:73, ack 3137, win 274, options [nop,nop,TS val 2689590080 ecr 2137928346], length 14
01:33:06.923296 IP 10.2.57.52.46851 > 37.48.83.75.6667: Flags [.], ack 4585, win 296, options [nop,nop,TS val 2689590120 ecr 2137928346], length 0
01:33:07.063449 IP 37.48.83.75.6667 > 10.2.57.52.46851: Flags [P.], seq 4585:5810, ack 59, win 227, options [nop,nop,TS val 2137928392 ecr 2689590079], length 1225
01:33:07.064127 IP 10.2.57.52.46851 > 37.48.83.75.6667: Flags [.], ack 5810, win 319, options [nop,nop,TS val 2689590260 ecr 2137928392], length 0
01:33:07.100864 IP 37.48.83.75.6667 > 10.2.57.52.46851: Flags [.], ack 73, win 227, options [nop,nop,TS val 2137928402 ecr 2689590080], length 0
```
What I see in irssi:
```
02:11 -!- Irssi: Looking up irc.freenode.net
02:11 -!- Irssi: Connecting to irc.freenode.net [192.186.157.43] port 6667
02:11 -!- Irssi: Connection to irc.freenode.net established
02:11 !tepper.freenode.net *** Looking up your hostname...
02:11 !tepper.freenode.net *** Checking Ident
02:11 !tepper.freenode.net *** Couldn't look up your hostname
02:12 !tepper.freenode.net *** No Ident response
02:12 -!- Welcome to the freenode Internet Relay Chat Network wani
...
...
```

Scenario2: (http-proxy configured in irssi and no forward rule on gateway):
```
iptables -D FORWARD -i eth1 -p tcp -m tcp --dport 6667 -j ACCEPT
```
```
settings = {
  core = { 
    real_name = "Nehal J Wani";
    user_name = "wani";
    nick = "wani";
    use_proxy = "yes";
    proxy_address = "$fqdn-of-proxy";
    proxy_port = "8080";
    proxy_string = "CONNECT %s:%d HTTP/1.0\012\012";
  };
  "fe-text" = { actlist_sort = "refnum"; };
};
```
tcpdump:
```
[root@proxy ~]# tcpdump host 10.2.57.52 -nn
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
listening on eth1, link-type EN10MB (Ethernet), capture size 65535 bytes
01:33:37.686295 IP 10.2.57.52.65152 > 65.55.138.112.80: Flags [S], seq 818253075, win 8192, options [mss 1460,nop,wscale 8,nop,nop,sackOK], length 0
01:33:37.686345 IP 10.4.8.204 > 10.2.57.52: ICMP 65.55.138.112 tcp port 80 unreachable, length 60
01:33:39.263946 IP 10.2.57.52.44937 > 10.4.8.204.8080: Flags [S], seq 3880370720, win 29200, options [mss 1460,sackOK,TS val 2689622461 ecr 0,nop,wscale 7], length 0
01:33:39.263968 IP 10.4.8.204.8080 > 10.2.57.52.44937: Flags [S.], seq 2981001349, ack 3880370721, win 14480, options [mss 1460,sackOK,TS val 2309737563 ecr 2689622461,nop,wscale 8], length 0
01:33:39.264208 IP 10.2.57.52.44937 > 10.4.8.204.8080: Flags [.], ack 1, win 229, options [nop,nop,TS val 2689622461 ecr 2309737563], length 0
01:33:39.264265 IP 10.2.57.52.44937 > 10.4.8.204.8080: Flags [P.], seq 1:43, ack 1, win 229, options [nop,nop,TS val 2689622461 ecr 2309737563], length 42
01:33:39.264276 IP 10.4.8.204.8080 > 10.2.57.52.44937: Flags [.], ack 43, win 57, options [nop,nop,TS val 2309737564 ecr 2689622461], length 0
01:33:39.264543 IP 10.2.57.52.44937 > 10.4.8.204.8080: Flags [P.], seq 43:103, ack 1, win 229, options [nop,nop,TS val 2689622461 ecr 2309737564], length 60
01:33:39.264554 IP 10.4.8.204.8080 > 10.2.57.52.44937: Flags [.], ack 103, win 57, options [nop,nop,TS val 2309737564 ecr 2689622461], length 0
01:33:39.553152 IP 10.4.8.204.8080 > 10.2.57.52.44937: Flags [P.], seq 1:40, ack 103, win 57, options [nop,nop,TS val 2309737853 ecr 2689622461], length 39
01:33:39.553598 IP 10.2.57.52.44937 > 10.4.8.204.8080: Flags [.], ack 40, win 229, options [nop,nop,TS val 2689622750 ecr 2309737853], length 0
01:33:39.743065 IP 10.4.8.204.8080 > 10.2.57.52.44937: Flags [P.], seq 40:103, ack 103, win 57, options [nop,nop,TS val 2309738043 ecr 2689622750], length 63
01:33:39.743912 IP 10.2.57.52.44937 > 10.4.8.204.8080: Flags [.], ack 103, win 229, options [nop,nop,TS val 2689622940 ecr 2309738043], length 0
01:33:39.921904 IP 10.4.8.204.8080 > 10.2.57.52.44937: Flags [P.], seq 103:219, ack 103, win 57, options [nop,nop,TS val 2309738221 ecr 2689622940], length 116
01:33:39.923755 IP 10.2.57.52.44937 > 10.4.8.204.8080: Flags [.], ack 219, win 229, options [nop,nop,TS val 2689623120 ecr 2309738221], length 0
01:33:40.701559 IP 10.2.57.52.65152 > 65.55.138.112.80: Flags [S], seq 818253075, win 8192, options [mss 1460,nop,wscale 8,nop,nop,sackOK], length 0
01:33:40.701580 IP 10.4.8.204 > 10.2.57.52: ICMP 65.55.138.112 tcp port 80 unreachable, length 60
01:33:46.701578 IP 10.2.57.52.65152 > 65.55.138.112.80: Flags [S], seq 818253075, win 8192, options [mss 1460,nop,nop,sackOK], length 0
01:33:46.701602 IP 10.4.8.204 > 10.2.57.52: ICMP 65.55.138.112 tcp port 80 unreachable, length 56
01:33:47.792358 IP 10.4.8.204.8080 > 10.2.57.52.44937: Flags [P.], seq 219:272, ack 103, win 57, options [nop,nop,TS val 2309746092 ecr 2689623120], length 53
01:33:47.792897 IP 10.2.57.52.44937 > 10.4.8.204.8080: Flags [.], ack 272, win 229, options [nop,nop,TS val 2689630990 ecr 2309746092], length 0
01:33:58.717174 IP 10.2.57.52.65153 > 134.170.165.249.80: Flags [S], seq 2452128696, win 8192, options [mss 1460,nop,wscale 8,nop,nop,sackOK], length 0
01:33:58.717198 IP 10.4.8.204 > 10.2.57.52: ICMP 134.170.165.249 tcp port 80 unreachable, length 60
01:34:01.717214 IP 10.2.57.52.65153 > 134.170.165.249.80: Flags [S], seq 2452128696, win 8192, options [mss 1460,nop,wscale 8,nop,nop,sackOK], length 0
01:34:01.717255 IP 10.4.8.204 > 10.2.57.52: ICMP 134.170.165.249 tcp port 80 unreachable, length 60
01:34:07.732622 IP 10.2.57.52.65153 > 134.170.165.249.80: Flags [S], seq 2452128696, win 8192, options [mss 1460,nop,nop,sackOK], length 0
01:34:07.732670 IP 10.4.8.204 > 10.2.57.52: ICMP 134.170.165.249 tcp port 80 unreachable, length 56
01:34:17.198834 IP 10.4.8.204.8080 > 10.2.57.52.44937: Flags [P.], seq 272:327, ack 103, win 57, options [nop,nop,TS val 2309775498 ecr 2689630990], length 55
01:34:17.201542 IP 10.2.57.52.44937 > 10.4.8.204.8080: Flags [.], ack 327, win 229, options [nop,nop,TS val 2689660398 ecr 2309775498], length 0
01:34:17.293026 IP 10.4.8.204.8080 > 10.2.57.52.44937: Flags [F.], seq 327, ack 103, win 57, options [nop,nop,TS val 2309775593 ecr 2689660398], length 0
01:34:17.293615 IP 10.2.57.52.44937 > 10.4.8.204.8080: Flags [F.], seq 103, ack 328, win 229, options [nop,nop,TS val 2689660491 ecr 2309775593], length 0
01:34:17.293630 IP 10.4.8.204.8080 > 10.2.57.52.44937: Flags [.], ack 104, win 57, options [nop,nop,TS val 2309775593 ecr 2689660491], length 0
01:34:19.736239 IP 10.2.57.52.65154 > 65.55.138.114.80: Flags [S], seq 2145044182, win 8192, options [mss 1460,nop,wscale 8,nop,nop,sackOK], length 0
01:34:19.736287 IP 10.4.8.204 > 10.2.57.52: ICMP 65.55.138.114 tcp port 80 unreachable, length 60
01:34:22.747974 IP 10.2.57.52.65154 > 65.55.138.114.80: Flags [S], seq 2145044182, win 8192, options [mss 1460,nop,wscale 8,nop,nop,sackOK], length 0
01:34:22.748008 IP 10.4.8.204 > 10.2.57.52: ICMP 65.55.138.114 tcp port 80 unreachable, length 60
01:34:28.747834 IP 10.2.57.52.65154 > 65.55.138.114.80: Flags [S], seq 2145044182, win 8192, options [mss 1460,nop,nop,sackOK], length 0
01:34:28.747886 IP 10.4.8.204 > 10.2.57.52: ICMP 65.55.138.114 tcp port 80 unreachable, length 56
01:34:40.747990 IP 10.2.57.52.65155 > 134.170.165.251.80: Flags [S], seq 2417415789, win 8192, options [mss 1460,nop,wscale 8,nop,nop,sackOK], length 0
01:34:40.748039 IP 10.4.8.204 > 10.2.57.52: ICMP 134.170.165.251 tcp port 80 unreachable, length 60
01:34:43.747732 IP 10.2.57.52.65155 > 134.170.165.251.80: Flags [S], seq 2417415789, win 8192, options [mss 1460,nop,wscale 8,nop,nop,sackOK], length 0
01:34:43.747781 IP 10.4.8.204 > 10.2.57.52: ICMP 134.170.165.251 tcp port 80 unreachable, length 60
01:34:49.763210 IP 10.2.57.52.65155 > 134.170.165.251.80: Flags [S], seq 2417415789, win 8192, options [mss 1460,nop,nop,sackOK], length 0
01:34:49.763265 IP 10.4.8.204 > 10.2.57.52: ICMP 134.170.165.251 tcp port 80 unreachable, length 56
```

What I see in irssi:
```
02:16 -!- Irssi: Looking up irc.freenode.net
02:16 -!- Irssi: Connecting to irc.freenode.net [10.4.8.204] port 6667
02:16 -!- Irssi: Connection to irc.freenode.net established
02:16 -!- HTTP/1.0 200 Connection established
02:16 !sinisalo.freenode.net *** Looking up your hostname...
02:16 !sinisalo.freenode.net *** Checking Ident
02:16 !sinisalo.freenode.net *** Couldn't look up your hostname
02:16 !sinisalo.freenode.net *** No Ident response
02:17 -!- ERROR Closing Link: 127.0.0.1 (Connection timed out)
02:17 -!- Irssi: Connection lost to irc.freenode.net
```

My conclusion:
With proxy settings set, irssi sends CONNECT requests to irc servers via proxy, but the SYN requests to port 80 are not being send through proxy. Hence the connection fails.

When I removed all proxy settings and used proxychains, there was no problem. So, I guess, there is no issue with squid or my network configuration. But there is some issue with irssi. Hence I am opening this issue.

With proxychains, the tcpdump says:
```
[root@proxy ~]# tcpdump host 10.2.57.52 -nn
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
listening on eth1, link-type EN10MB (Ethernet), capture size 65535 bytes
02:14:11.138457 IP 10.2.57.52.65287 > 65.55.138.112.80: Flags [S], seq 4152375060, win 8192, options [mss 1460,nop,wscale 8,nop,nop,sackOK], length 0
02:14:11.138515 IP 10.4.8.204 > 10.2.57.52: ICMP 65.55.138.112 tcp port 80 unreachable, length 60
02:14:14.138335 IP 10.2.57.52.65287 > 65.55.138.112.80: Flags [S], seq 4152375060, win 8192, options [mss 1460,nop,wscale 8,nop,nop,sackOK], length 0
02:14:14.138362 IP 10.4.8.204 > 10.2.57.52: ICMP 65.55.138.112 tcp port 80 unreachable, length 60
02:14:20.138525 IP 10.2.57.52.65287 > 65.55.138.112.80: Flags [S], seq 4152375060, win 8192, options [mss 1460,nop,nop,sackOK], length 0
02:14:20.138548 IP 10.4.8.204 > 10.2.57.52: ICMP 65.55.138.112 tcp port 80 unreachable, length 56
02:14:32.153940 IP 10.2.57.52.65288 > 207.46.114.58.80: Flags [S], seq 1844392939, win 8192, options [mss 1460,nop,wscale 8,nop,nop,sackOK], length 0
02:14:32.153959 IP 10.4.8.204 > 10.2.57.52: ICMP 207.46.114.58 tcp port 80 unreachable, length 60
02:14:35.169055 IP 10.2.57.52.65288 > 207.46.114.58.80: Flags [S], seq 1844392939, win 8192, options [mss 1460,nop,wscale 8,nop,nop,sackOK], length 0
02:14:35.169075 IP 10.4.8.204 > 10.2.57.52: ICMP 207.46.114.58 tcp port 80 unreachable, length 60
02:14:39.768902 IP 10.2.57.52.44951 > 10.4.8.204.8080: Flags [S], seq 983454117, win 29200, options [mss 1460,sackOK,TS val 2692082998 ecr 0,nop,wscale 7], length 0
02:14:39.768922 IP 10.4.8.204.8080 > 10.2.57.52.44951: Flags [S.], seq 4227999916, ack 983454118, win 14480, options [mss 1460,sackOK,TS val 2312198068 ecr 2692082998,nop,wscale 8], length 0
02:14:39.769280 IP 10.2.57.52.44951 > 10.4.8.204.8080: Flags [.], ack 1, win 229, options [nop,nop,TS val 2692082998 ecr 2312198068], length 0
02:14:39.769437 IP 10.2.57.52.44951 > 10.4.8.204.8080: Flags [P.], seq 1:40, ack 1, win 229, options [nop,nop,TS val 2692082999 ecr 2312198068], length 39
02:14:39.769447 IP 10.4.8.204.8080 > 10.2.57.52.44951: Flags [.], ack 40, win 57, options [nop,nop,TS val 2312198069 ecr 2692082999], length 0
02:14:40.008692 IP 10.4.8.204.8080 > 10.2.57.52.44951: Flags [P.], seq 1:40, ack 40, win 57, options [nop,nop,TS val 2312198308 ecr 2692082999], length 39
02:14:40.010237 IP 10.2.57.52.44951 > 10.4.8.204.8080: Flags [.], ack 40, win 229, options [nop,nop,TS val 2692083239 ecr 2312198308], length 0
02:14:40.010402 IP 10.2.57.52.44951 > 10.4.8.204.8080: Flags [P.], seq 40:51, ack 40, win 229, options [nop,nop,TS val 2692083240 ecr 2312198308], length 11
02:14:40.010410 IP 10.4.8.204.8080 > 10.2.57.52.44951: Flags [.], ack 51, win 57, options [nop,nop,TS val 2312198310 ecr 2692083240], length 0
02:14:40.010864 IP 10.2.57.52.44951 > 10.4.8.204.8080: Flags [P.], seq 51:98, ack 40, win 229, options [nop,nop,TS val 2692083240 ecr 2312198310], length 47
02:14:40.010873 IP 10.4.8.204.8080 > 10.2.57.52.44951: Flags [.], ack 98, win 57, options [nop,nop,TS val 2312198310 ecr 2692083240], length 0
02:14:40.195729 IP 10.4.8.204.8080 > 10.2.57.52.44951: Flags [P.], seq 40:106, ack 98, win 57, options [nop,nop,TS val 2312198495 ecr 2692083240], length 66
02:14:40.236324 IP 10.2.57.52.44951 > 10.4.8.204.8080: Flags [.], ack 106, win 229, options [nop,nop,TS val 2692083466 ecr 2312198495], length 0
02:14:40.413858 IP 10.4.8.204.8080 > 10.2.57.52.44951: Flags [P.], seq 106:159, ack 98, win 57, options [nop,nop,TS val 2312198713 ecr 2692083466], length 53
02:14:40.414258 IP 10.2.57.52.44951 > 10.4.8.204.8080: Flags [.], ack 159, win 229, options [nop,nop,TS val 2692083643 ecr 2312198713], length 0
02:14:41.169120 IP 10.2.57.52.65288 > 207.46.114.58.80: Flags [S], seq 1844392939, win 8192, options [mss 1460,nop,nop,sackOK], length 0
02:14:41.169161 IP 10.4.8.204 > 10.2.57.52: ICMP 207.46.114.58 tcp port 80 unreachable, length 56
02:14:41.360161 IP 10.4.8.204.8080 > 10.2.57.52.44951: Flags [P.], seq 159:228, ack 98, win 57, options [nop,nop,TS val 2312199660 ecr 2692083643], length 69
02:14:41.360873 IP 10.2.57.52.44951 > 10.4.8.204.8080: Flags [.], ack 228, win 229, options [nop,nop,TS val 2692084590 ecr 2312199660], length 0
02:14:46.266424 IP 10.4.8.204.8080 > 10.2.57.52.44951: Flags [.], seq 228:3124, ack 98, win 57, options [nop,nop,TS val 2312204566 ecr 2692084590], length 2896
02:14:46.266441 IP 10.4.8.204.8080 > 10.2.57.52.44951: Flags [.], seq 3124:6020, ack 98, win 57, options [nop,nop,TS val 2312204566 ecr 2692084590], length 2896
02:14:46.266586 IP 10.4.8.204.8080 > 10.2.57.52.44951: Flags [P.], seq 6020:6076, ack 98, win 57, options [nop,nop,TS val 2312204566 ecr 2692084590], length 56
02:14:46.267066 IP 10.2.57.52.44951 > 10.4.8.204.8080: Flags [.], ack 1676, win 251, options [nop,nop,TS val 2692089496 ecr 2312204566], length 0
02:14:46.267080 IP 10.2.57.52.44951 > 10.4.8.204.8080: Flags [.], ack 3124, win 274, options [nop,nop,TS val 2692089496 ecr 2312204566], length 0
02:14:46.267160 IP 10.2.57.52.44951 > 10.4.8.204.8080: Flags [.], ack 4572, win 296, options [nop,nop,TS val 2692089497 ecr 2312204566], length 0
02:14:46.267308 IP 10.2.57.52.44951 > 10.4.8.204.8080: Flags [.], ack 6076, win 320, options [nop,nop,TS val 2692089497 ecr 2312204566], length 0
02:14:46.267322 IP 10.2.57.52.44951 > 10.4.8.204.8080: Flags [P.], seq 98:112, ack 6076, win 320, options [nop,nop,TS val 2692089497 ecr 2312204566], length 14
02:14:46.267329 IP 10.4.8.204.8080 > 10.2.57.52.44951: Flags [.], ack 112, win 57, options [nop,nop,TS val 2312204567 ecr 2692089497], length 0
02:14:46.451040 IP 10.4.8.204.8080 > 10.2.57.52.44951: Flags [P.], seq 6076:6738, ack 112, win 57, options [nop,nop,TS val 2312204751 ecr 2692089497], length 662
02:14:46.491259 IP 10.2.57.52.44951 > 10.4.8.204.8080: Flags [.], ack 6738, win 343, options [nop,nop,TS val 2692089721 ecr 2312204751], length 0
02:14:49.376751 IP 10.2.57.52.44951 > 10.4.8.204.8080: Flags [P.], seq 112:140, ack 6738, win 343, options [nop,nop,TS val 2692092606 ecr 2312204751], length 28
02:14:49.376767 IP 10.4.8.204.8080 > 10.2.57.52.44951: Flags [.], ack 140, win 57, options [nop,nop,TS val 2312207676 ecr 2692092606], length 0
02:14:51.508543 IP 10.4.8.204.8080 > 10.2.57.52.44951: Flags [P.], seq 6738:6812, ack 140, win 57, options [nop,nop,TS val 2312209808 ecr 2692092606], length 74
02:14:51.508904 IP 10.2.57.52.44951 > 10.4.8.204.8080: Flags [.], ack 6812, win 343, options [nop,nop,TS val 2692094738 ecr 2312209808], length 0
02:14:53.169290 IP 10.2.57.52.65289 > 134.170.165.248.80: Flags [S], seq 2850542057, win 8192, options [mss 1460,nop,wscale 8,nop,nop,sackOK], length 0
02:14:53.169316 IP 10.4.8.204 > 10.2.57.52: ICMP 134.170.165.248 tcp port 80 unreachable, length 60
02:14:56.168784 IP 10.2.57.52.65289 > 134.170.165.248.80: Flags [S], seq 2850542057, win 8192, options [mss 1460,nop,wscale 8,nop,nop,sackOK], length 0
02:14:56.168805 IP 10.4.8.204 > 10.2.57.52: ICMP 134.170.165.248 tcp port 80 unreachable, length 60
02:15:02.168813 IP 10.2.57.52.65289 > 134.170.165.248.80: Flags [S], seq 2850542057, win 8192, options [mss 1460,nop,nop,sackOK], length 0
02:15:02.168841 IP 10.4.8.204 > 10.2.57.52: ICMP 134.170.165.248 tcp port 80 unreachable, length 56
```

What I see in irssi:
```
02:21 -!- Irssi: Looking up irc.freenode.net
02:21 -!- Irssi: Connecting to irc.freenode.net [91.217.189.42] port 6667
02:21 -!- Irssi: Connection to irc.freenode.net established
02:21 !sinisalo.freenode.net *** Looking up your hostname...
02:21 !sinisalo.freenode.net *** Checking Ident
02:21 !sinisalo.freenode.net *** Couldn't look up your hostname
02:21 !sinisalo.freenode.net *** No Ident response
02:21 -!- Welcome to the freenode Internet Relay Chat Network wani
...
...
```

**Comment by dequis** [comment]
>but the SYN requests to port 80 are not being send through proxy

The only port 80 requests I see there are IP addresses in microsoft ranges, apparently used for windows update.

Also tcpdump without packet contents is kinda terrible to debug this stuff. It's looking at the problem in the wrong layer.

TL;DR of this issue that http proxy support is broken with this squid or this connection string. The tcpdump and iptables stuff can be ignored.

**Comment by ailin-nemui** [comment]
@nehaljwani please obtain a tcpdump with package content of the TCP stream which contains the IRC connection if this is still an issue

**Comment by ailin-nemui** [comment]
please reopen with more info if this is still an issue
